editor/components/autosave-monitor/index.js

```diff
@@ -6,10 +6,13 @@ import { withSelect, withDispatch } from '@wordpress/data';
 
 export class AutosaveMonitor extends Component {
 	componentDidUpdate( prevProps ) {
-		const { isDirty, isSaveable } = this.props;
-		if ( prevProps.isDirty !== isDirty ||
-				prevProps.isSaveable !== isSaveable ) {
-			this.toggleTimer( isDirty && isSaveable );
+		const { isDirty, isSaveable, isAutosaveable } = this.props;
+		if (
+			prevProps.isDirty !== isDirty ||
+			prevProps.isSaveable !== isSaveable ||
+			prevProps.isAutosaveable !== isAutosaveable
+		) {
+			this.toggleTimer( isDirty && isSaveable && isAutosaveable );
```

Comment:
I seem to recall raising previously, but am unable to find the conversation and resulting rebuttal (if one exists): When would `isAutosaveable` be `true` and `isSaveable` be `false` ? The issue I'm getting to here is that it seems like `isSaveable` should be built in to the selector logic for `isAutosaveable` and thus made redundant as a separate prop here, simplifying to `isDirty && isAutosaveable`.

Reply:
In this case `isSaveable ` maps to `isEditedPostSaveable` which tests if the post contains a title, an excerpt, or non-empty content. if you edit the content of an existing post and remove everything, the post is no longer savable, this prevents calling the autosave in that case. its possibly redundant to have this here and in `effects.js`, should i remove it there?

Reply:
Should `isEditedPostSaveable` be tested as part of the `isAutosaveable` selector? In other words, is it _ever_ the case that an autosave is valid where `isEditedPostSaveable` would otherwise return `false` ? This would allow us to simplify both the effect and this component, where we could simply test `isAutosaveable` and be assured that the post is also "saveable" (with valid content).

Reply:
sure, thats a good idea. I can move that test into `isAutosavable` to clear up the logic here.

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/6257/commits/7b4bb191500c6bc81f372c75d6937d6472d0e24c, adding the `isEditedPostSaveable` logic to `isPostAutosaveable` and removing the extra calls to `isEditedPostSaveable` here